pkg/importer/spanner/sysl.arrai

```diff
@@ -0,0 +1,138 @@
+# size returns the size of an attribute
+let size = \length
+    cond {
+        length = 'MAX': '',
+        //seq.has_prefix('0x', length): '(' ++ //seq.trim_prefix('0x', length) ++ ')',
+        length > 0: $`(${length})`,
+        _: '',
+    };
+
+# attributePatternToken returns the token based on attribute value
+let attributePatternToken = \entity \attr \token
+    cond {
+        (attr.options?:{}): token,
+        (entity.primary_key where //seq.contains(attr.name, .)): token,
+        let fks = (entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
```

Comment:
You do a lot of work to build this data structure, and them merely check whether it is non-empty. Is there a simpler way to write this condition?

Reply:
attributePatternToken checks if the attr passed is included in a foreign key and return a token if included. Token is generally a pattern delimiter which needs to be rendered conditionally.

Reply:
simplified